### PR TITLE
PIM-6175: Remove deprecated methods

### PR DESCRIPTION
**Pull request by pierallard**
What do this PR ?

- It moves UserBundle repositories to right folders (all the deprecated ones)
- It changes a method visibility (public -> protected, deprecated too)
- It changes a "will be removed in 1.7" to "will be removed in 1.8" (too many work to do)
- It removes useless "use xxx" on headers
- It fixes the 2 failing behats

For info, the first failing behat is failing cause of EE permissions, I simply deactivated it in EE, because execute it in EE brings nothing.

### CODE REVIEW

File: src/Pim/Bundle/UserBundle/spec/Entity/Repository/UserRepositorySpec.php [hunk 1]
```diff
@@ -35,7 +35,7 @@ function let(
     function it_is_a_user_repository()
     {
         $this->shouldHaveType('Doctrine\ORM\EntityRepository');
-        $this->shouldHaveType('Pim\Bundle\UserBundle\Entity\Repository\UserRepositoryInterface');
+        $this->shouldHaveType('Pim\Bundle\UserBundle\Repository\UserRepositoryInterface');
```
Comment (nidup) [inline src/Pim/Bundle/UserBundle/spec/Entity/Repository/UserRepositorySpec.php]: Pim\Bundle\UserBundle\Doctrine\ORM\Repository\UserRepository

**Comment by a2xchip** [comment]
As a cool reviewer... :trolleybus: 
When family is saved by user with limited access to attributes, all filtered attributes are being removed. Please unfilter before updating the family ;-) Rest\FamilyController missed you. 

Overall PR is amazing!! :heart:

**Comment by pierallard** [comment]
@a2xchip Ticket is created for this issue.

**Comment by SamirBoulil** [comment]
Seems great, but you add some Business code in the FamilyNormalizer ? Why is this made in this PR

**Comment by pierallard** [comment]
@SamirBoulil It's for the behat fix

**Pull request merged**